feature(notes): added latest sep 7 meeting notes

TheLarkInn opened the pull request:
@webpack/collab-team 
@webpack/contributor-team 

Alright 👀 appreciated for the review of the notes. Thank you

TheLarkInn committed:
feature(notes): added latest sep 7 meeting notes

TheLarkInn committed:
Update september-07.md

reference correct details

bebraw commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,82 @@
+
+	* Don't want linting too strict, just enough to provide, autoformatting, and eliminate the number of round of reviews for PR's.
+
+* The topic of Cont. Deployment came up for our documentation. We veiw some other pages and it apeard Codeship is a popular option as well as Travis but requires more work. An issue will be created in the docs repo. 
```
/s/veiw/view

TheLarkInn replied:
Maybe I should setup proselint on here. 🤔

TheLarkInn committed:
Update september-07.md

TheLarkInn committed:
Update september-07.md

fixed indentation on view

TheLarkInn committed:
Update september-07.md

e-cloud commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,80 @@
+## August 31 ([discuss](https://github.com/webpack/meeting-notes/pull/13))
```
date isn't right

TheLarkInn replied:
Thank you!

DanielRosenwasser commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,80 @@
+	* Johannes raised this issue because it is notable that some scaled builds take long to finish (even if it is 2-3 minutes). Johannes mentioned [`HappyPack`](https://github.com/amireh/happypack) is a common package that allows you to multithread the loader process. However there are struggles with the way that "hacky" loaders work that make `HappyPack` impossible for `awesome-typescript-loader` a ...
+	* Sean describes this "hackyness" is when a loader accesses `compiler` properties from `this` such as `this._compiler` to execute plugin based functionality. Needs to be deprecated.
+	* Tobias mentions that maybe we can add functionality to the loader API to allow this to work. Sean explained that the reason these TS loaders need plugin access is because it needs to defer typechecking until the `compilation` is finished. 
+	* Tobias mentions that deferred execution could be a feature we need to add to the loader API. Could also be useful for image sprites, etc. 
```
:100: :+1: :+1: :tada:

mxstbr commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,80 @@
+* We discussed between `webpack.js.org` and `webpack.github.io`. After we did the research we decided that we would setup `webpack.js.org`. Tobias is going to tackle this a bit later today. 
+
+#### Analytics 
+* Sean asked whether or not we have page analyitcs on our docs site currently. Tobias gave access to all the analytic data. Sean also brought up the idea of having webpack the tool ask if users would let us submit analyitcs data. Tabeled discussion as its a severe nice-to-have.
```
Just warning you: Beware of the shit storm, make it very obvious that it's opt-outable. (see this very long discussion: https://github.com/atom/atom/issues/4966)

jhnns replied:
The tool would be opt-in

jhnns commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,80 @@
+		normal phase:
+		* store dependencies, hash of dependencies and source in disk cache (edited)
+		* alternativly it could use loader-runner to run the remaining loaders. This would allow to cache more data: emitted files, errors, warnings. This would not be possible in the normal loader chain.
+	* Issues will be created for each of these missing features. (Sean).
```
There's one issue/defect missing here: We are planing to deprecate all properties and methods that make a multi-thread compilation impossible, that is:
- No synchronous APIs (`resolveSync`)
- No shared state (`_compiler`)

Could you add that @TheLarkInn ?

TheLarkInn replied:
Yes I will add this.

TheLarkInn replied:
Updated below. Let me know if this is what you are referring to.

TheLarkInn committed:
Update september-07.md

Update correct date and comments from Johannes on mentioning deprecation warnings.

timmfin commented on 2016-09/september-07.md:
```diff
@@ -0,0 +1,87 @@
+	* additionally noted it would fall after optimization and before id creation (in the compilation step)
+	* This would also allow ts-loader and awesome-typescript-loader to execute typechecking in the resolve promise. (allowing tools like HappyPack to execute loaders parallel/multithreaded)
+	* Johannes mentions that it would be important that we get feedback from loader authors to see if this would help solve some of their issues (including HappyPack).
+	* Tobias described, (once implementing this feature [based upon the comments in the linked issue]) that a `disk-cache-loader` would look similar to having the following.
```
If it is of interest at all, I have a local prototype of something very similar to a `disk-cache-loader`. A comment from my source that describes what I was trying to do:

``` js
  // If we have a persisted cache for this loader run,
  //   and that hash hasn't changed since last time (note hash includes file content, full loader request, version, and context)
  //   and none of its dependencies have changed (pulled from previous `module.fileDependencies`),
  // then we can return the cached result & source map from the LRU-ed disk cache
```

It still is quite rough, and I'm very aware that "misbehaving" plugins and loaders could break it (for our use case I'm ok to whitelist the cache for specific loaders). But I'm glad to try and get it to a share-able state in the future if the core team thinks that is helpful.

Jessidhia replied:
One potential issue is with plugins that use an external config; e.g. eslint, babel (with .babelrc), ts (tsconfig.json is pretty much required). If the config changes, the output will most likely change too, but a naive approach would only notice it if it's part of the query.

Other loaders have the (much more convenient) option of reading their config from the webpack config object itself, like postcss-loader. As the config option may be a function, it can be harder to cache (in fact I use that property to use a different postcss syntax plugin per file type).

Perhaps a caching API could be provided, where the plugin itself can declare the state it is going to use to execute its load?

This would still leave funny cases such as browserslist files.

timmfin replied:
Even though plugins don't have a dedicated API for it, they can still append config files to the [compiler's list of `fileDependencies`](https://github.com/webpack/webpack-dev-server/issues/34#issuecomment-47420992) which will cause the watcher to track when that config file changes. Plus this is even easier for loaders since there is [dedicated `addDependency` API](https://webpack.github.io/docs/loaders.html#adddependency).

Sure, there still is trickiness boiling that config object down and persisting the state in some hash (libs like [this](https://github.com/puleos/object-hash) can help?) . And having dedicated webpack APIs to modify/append a compilation's hash or individual module hashes would be helpful. But I think that is straight-forward.

(IMHO, the hard part is teaching and encouraging loader/plugin authors to remember to track extra things like config files, a version number to change the hash as internal code changes, etc).

jhnns replied:
Maybe you could give feedback on [this proposal](https://github.com/webpack/webpack/issues/2975)?
